### PR TITLE
feat: add valid_parentheses.cpp

### PR DESCRIPTION
**Pull request by tuan-dang-0**
#### Description of Change

Added stack implementation for a solution to valid parentheses in "others"

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
Added valid_parentheses.cpp to "others"

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1896"></a>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,5 @@ a.out
 *.app
 
 build/
+/.vs/C-Plus-Plus/v16
+/.vs
```
Comment (Panquesito7) [inline .gitignore:39]: Please revert the changes in this file. Thank you.

File: others/valid_parentheses.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for cout
+#include <map>       /// for std::map
+#include <stack>     /// for std::stack
+
+/**
+ * @namespace others
+ * @brief other algorithms
```
Comment (Panquesito7) [inline others/valid_parentheses.cpp:24]:
```suggestion
 * @brief Other algorithms
```

File: others/valid_parentheses.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
+/**
+ * @file
+ * @brief Stack implementation to check Valid Parentheses in string
+ * (https://leetcode.com/problems/valid-parentheses/)
```
Comment (Panquesito7) [inline others/valid_parentheses.cpp:4]:
```suggestion
 * @brief [Stack implementation to check Valid Parentheses in string](https://leetcode.com/problems/valid-parentheses/)
```

File: others/valid_parentheses.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * @namespace others
+ * @brief other algorithms
+ */
+namespace others {
+
+/**
+ * @namespace valid_parentheses
+ * @brief Functions for implementation
```
Comment (Panquesito7) [inline others/valid_parentheses.cpp:30]:
```suggestion
 * @brief Functions for the [Check Valid Parentheses in string](https://leetcode.com/problems/valid-parentheses/) implementation
```

File: others/valid_parentheses.cpp [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            // If there is not an opening bracket to match current closing
+            // bracket, string invalid
+            else
+                return false;
+        }
+    }
+
+    // If there are open parentheses unmatched, string invalid
+    if (!stk.empty())
+        return false;
```
Comment (Panquesito7) [inline others/valid_parentheses.cpp:70]:
```suggestion
    if (!stk.empty()) {
        return false;
    }
```

File: others/valid_parentheses.cpp [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    std::cout << "All test cases passed!" << std::endl;
+}
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();  // run self-test implementations
+    return 0;
+}
```
Comment (Panquesito7) [inline others/valid_parentheses.cpp:100]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 🚀

**Comment by tuan-dang-0** [comment]
Hey! just added the changes you requested, thank you.

File: others/valid_parentheses.cpp [hunk 7]
```diff
@@ -0,0 +1,104 @@
+/**
+ * @file
+ * @brief [Stack implementation to check Valid Parentheses in string]
+   (https://leetcode.com/problems/valid-parentheses/)
```
Comment (Panquesito7) [inline others/valid_parentheses.cpp:4]:
```suggestion
 * (https://leetcode.com/problems/valid-parentheses/)
```

File: others/valid_parentheses.cpp [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+/**
+ * @namespace others
+ * @brief Other algorithms
+ */
+namespace others {
+
+/**
+ * @namespace valid_parentheses
+ * @brief Functions for the [Check Valid Parentheses in string]
+   (https://leetcode.com/problems/valid-parentheses/)
```
Comment (Panquesito7) [inline others/valid_parentheses.cpp:31]:
```suggestion
 * (https://leetcode.com/problems/valid-parentheses/) algorithm implementation
```

File: others/valid_parentheses.cpp [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    // input = "}", returns false
+    assert(others::valid_parentheses::is_valid("}") == false);
+    // input = "(}", returns false
+    assert(others::valid_parentheses::is_valid("(}") == false);
+    // input = "(){}[]", returns true
+    assert(others::valid_parentheses::is_valid("(){}[]") == true);
+    // input = "{[[]{}]}", returns true
+    assert(others::valid_parentheses::is_valid("{[[]{}]}") == true);
+
+    std::cout << "All test cases passed!" << std::endl;
```
Comment (Panquesito7) [inline others/valid_parentheses.cpp:95]:
```suggestion
    std::cout << "All test cases have successfully passed!" << std::endl;
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**